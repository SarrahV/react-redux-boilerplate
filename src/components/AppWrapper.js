import styled from 'styled-components';

const AppWrapper = styled.div`
  display: flex;
  flex-direction: column;
  color: DarkSlateGray;
  min-height: 100vh;
`;

export default AppWrapper;
