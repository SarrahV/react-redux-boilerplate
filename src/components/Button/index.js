import styled from 'styled-components';

const Button = styled.button`
  background: teal;
  border: none;
  color: ivory;
  padding: 1em;
`;

export default Button;
