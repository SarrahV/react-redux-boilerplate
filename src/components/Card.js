import React, { PropTypes } from 'react';
import styled from 'styled-components';

const Card = (props) => {
  const CardWrapper = styled.div`
    padding: 20px;
  `;

  return (
    <CardWrapper>
      {props.children}
    </CardWrapper>
  );
};

Card.propTypes = {
  children: PropTypes.shape,
};

export default Card;
