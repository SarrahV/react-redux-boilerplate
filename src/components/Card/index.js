import React, { PropTypes } from 'react';

import Wrapper from './Wrapper';

const Card = props =>
  <Wrapper>
    {props.children}
  </Wrapper>;

Card.propTypes = {
  children: PropTypes.element,
};

export default Card;
