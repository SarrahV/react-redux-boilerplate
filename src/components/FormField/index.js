import React, { PropTypes } from 'react';
import Feedback from './Feedback';
import Input from './Input';
import Label from './Label';
import Wrapper from './Wrapper';

const InputField = ({
  input,
  label,
  placeholder,
  type,
  meta: { touched, error, warning },
}) => {
  const showError = touched && error;
  const showWarning = touched && warning;
  return (
    <Wrapper>
      <Label
        error={showError}
        warning={showWarning}
      >
        {label}
      </Label>
      <Input
        error={showError}
        warning={showWarning}
        placeholder={placeholder}
        type={type}
        {...input}
      />
      {((showError &&
        <Feedback
          error={showError}
          warning={showWarning}
        >
          {error}
        </Feedback>) ||
      (showWarning &&
        <Feedback
          error={showError}
          warning={showWarning}
        >
          {warning}
        </Feedback>))}
    </Wrapper>
  );
};

InputField.propTypes = {
  input: PropTypes.shape.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  meta: PropTypes.shape({
    touched: PropTypes.boolean,
    error: PropTypes.string,
    warning: PropTypes.string,
  }),
};

export default InputField;
