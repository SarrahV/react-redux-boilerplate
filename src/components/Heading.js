import styled from 'styled-components';

const Heading = styled.h1`
  margin-top: 0;
`;

export default Heading;
