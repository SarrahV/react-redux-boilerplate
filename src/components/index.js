export AppWrapper from './AppWrapper';
export Card from './Card';
export Heading from './Heading';
export MainContent from './MainContent';
export Navigation from './Navigation';
export NotFound from './NotFound';
