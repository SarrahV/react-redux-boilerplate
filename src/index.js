import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { BrowserRouter, Match, Miss } from 'react-router';

import { MainContent, NotFound } from 'components';
import { App, Cats, Humans, FormWithCats, Home, Navigation } from 'containers';
import initializeStore from './redux/init';

import './globalStyles';

const store = initializeStore();

const Boilerplate = () => (
  <Provider store={store}>
    <BrowserRouter>
      <App>
        <Navigation />
        <MainContent>
          <Match exactly pattern="/" component={Home} />
          <Match exactly pattern="/cats" component={Cats} />
          <Match exactly pattern="/form" component={FormWithCats} />
          <Match exactly pattern="/humans" component={Humans} />
          <Miss component={NotFound} />
        </MainContent>
      </App>
    </BrowserRouter>
  </Provider>
);

ReactDOM.render(
  <Boilerplate />,
  document.getElementById('mount'),
);
