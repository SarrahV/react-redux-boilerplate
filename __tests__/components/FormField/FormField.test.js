import React from 'react';
import renderer from 'react-test-renderer';

import { FormField } from '../../../src/components';

describe('<FormField />', () => {
  it('renders a form field component', () => {
    const component = renderer.create(
      <FormField />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('renders a form field with a label', () => {
    const component = renderer.create(
      <FormField label="This field's label" />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('renders a form field with an error', () => {
    const component = renderer.create(
      <FormField
        error="You are bad at forms"
        touched
      />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('renders a form field with a warning', () => {
    const component = renderer.create(
      <FormField
        warning="You are kind of bad at forms"
        touched
      />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
