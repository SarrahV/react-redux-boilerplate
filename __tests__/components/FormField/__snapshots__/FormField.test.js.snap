exports[`<FormField /> renders a form field component 1`] = `
<div
  className="bzWcLi">
  <input
    className="iZQdaK"
    placeholder=""
    type="text" />
</div>
`;

exports[`<FormField /> renders a form field with a label 1`] = `
<div
  className="bzWcLi">
  <div
    className="bjhZVI">
    This field's label
  </div>
  <input
    className="iZQdaK"
    placeholder=""
    type="text" />
</div>
`;

exports[`<FormField /> renders a form field with a warning 1`] = `
<div
  className="bzWcLi">
  <input
    className="iZQdaK"
    placeholder=""
    type="text" />
</div>
`;

exports[`<FormField /> renders a form field with an error 1`] = `
<div
  className="bzWcLi">
  <input
    className="iZQdaK"
    placeholder=""
    type="text" />
</div>
`;
